GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,209 @@
+﻿using CommandLine;
+using GVFS.Common;
+using GVFS.Common.Database;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.NamedPipes;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.CommandLine
+{
+    [Verb(StatisticsVerb.StatisticsVerbName, HelpText = "Get statistics for the health state of the repository")]
+    public class StatisticsVerb : GVFSVerb.ForExistingEnlistment
+    {
+        private const string StatisticsVerbName = "statistics";
+
+        private const int DirectoryDisplayCount = 5;
+
+        [Option(
+            'd',
+            "directory",
+            Required = false,
+            HelpText = "Run the statistics tool on a specific directory")]
+        public string Directory { get; set; }
+
+        protected override string VerbName
+        {
+            get { return StatisticsVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.DotGitRoot.Substring(0, enlistment.DotGitRoot.Length - 4);
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);

```

Comment:
It looks like you're trying to extract a relative path here, do we need code to handle a user passing in an absolute path (e.g. `C:\Repos\VFSForGit\src\GVFS` or `/Users/wilbaker/Repos/VFSForGit/src/GVFS`)?

Reply:
Yeah those types of paths definitely need some work. See the comment above on paths relative to `src`

Reply:
Also for the above feedback (I'm not sure why it won't let me respond to it via a comment) I think `targetDirectory` is probably a better name for it. It functions as a filter so that statistics are only calculated based on paths which fall underneath it in the tree